src/Gui/PythonEditor.cpp

```diff
@@ -256,6 +265,69 @@ void PythonEditor::onUncomment()
     cursor.endEditBlock();
 }
 
+void PythonEditor::onExecuteInConsole()
+{
+    QTextCursor cursor = textCursor();
+    int selStart = cursor.selectionStart();
+    int selEnd = cursor.selectionEnd();
+    QTextBlock block;
+    QString selectedCode;
+
+    for (block = document()->begin(); block.isValid(); block = block.next()) {
+        int pos = block.position();
+        int off = block.length() - 1;
+        if (pos >= selStart || pos + off >= selStart) {
+            if (pos + 1 > selEnd)
+                break;
+
+            QString lineText = block.text();
+            selectedCode.append(lineText + QLatin1String("\n"));
+        }
+    }

```

Comment:
This could be a private function in `PythonEditor` - even if it would be used only once it would be at least properly named.